prow/cmd/branchprotector/protect.go

```diff
@@ -267,18 +274,84 @@ func (p *protector) UpdateRepo(orgName string, repoName string, repo config.Repo
 		}
 	}
 
+	var collaborators, teams []string
+	if p.verifyRestrictions {
+		collaborators, err = p.authorizedCollaborators(orgName, repoName)
+		if err != nil {
+			logrus.Infof("%s/%s: error getting list of collaborators: %v", orgName, repoName, err)
+			return err
+		}
+
+		teams, err = p.authorizedTeams(orgName, repoName)
+		if err != nil {
+			logrus.Infof("%s/%s: error getting list of teams: %v", orgName, repoName, err)
+			return err
+		}
+	}
+
 	for bn, githubBranch := range branches {
 		if branch, err := repo.GetBranch(bn); err != nil {
 			return fmt.Errorf("get %s: %v", bn, err)
-		} else if err = p.UpdateBranch(orgName, repoName, bn, *branch, githubBranch.Protected); err != nil {
+		} else if err = p.UpdateBranch(orgName, repoName, bn, *branch, githubBranch.Protected, collaborators, teams); err != nil {
 			return fmt.Errorf("update %s from protected=%t: %v", bn, githubBranch.Protected, err)
 		}
 	}
 	return nil
 }
 
+// authorizedCollaborators returns the list of Logins for users that are
+// authorized to write to a repository.
+func (p *protector) authorizedCollaborators(org, repo string) ([]string, error) {
+	collaborators, err := p.client.ListCollaborators(org, repo)
+	if err != nil {
+		return nil, err
+	}
+	var authorized []string
+	for _, c := range collaborators {
+		if c.Permissions.Admin || c.Permissions.Push {
+			authorized = append(authorized, c.Login)
```

Comment:
Do we need to use `prowgithub.NormLogin()` here?

Reply:
Thanks. I have updated the PR. From reading the github API doc, it does not seem that `Login` field could include `@`, but perhaps this change makes it more straightforward to write the config file, i.e. use lower case and no `@`.

Reply:
It is more that `BenTheElder` may or may not be equivalent to `bentheelder`.